argoexec/tests/test_rock.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Tests for required artifacts to be present in the ROCK image.
+#
+
+from charmed_kubeflow_chisme.rock import CheckRock
+from pathlib import Path
+
+import os
+import logging
+import random
+import pytest
+import string
+import subprocess
+import yaml
+from pytest_operator.plugin import OpsTest
+
+@pytest.fixture()
+def rock_test_env():
+    """Yelds random docker container name, then cleans it up after."""
+    container_name = "".join([str(i) for i in random.choices(string.ascii_lowercase, k=8)])
+    yield container_name
+
+    try:
+        subprocess.run(["docker", "rm", container_name], check=True, capture_output=True)
```

Comment:
There is a docker [Python SDK](https://docker-py.readthedocs.io/en/stable/), maybe you want to explore that option

Reply:
The issue is the size of the dependencies needed for testing. Last time we introduced Chisme for testing and it was rejected, because it was a large dependency. IMO the same rationale applies to this Python Docker SDK.

Reply:
I don't think that rationale applies here, as this is in fact, a Python script, and not a command that runs inside a tox file. It's up to you if you want to use the SDK, it was a suggestion to make this script cleaner.